cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt

```diff
@@ -0,0 +1,145 @@
+package com.luxoft.blockchainlab.corda.hyperledger.indy
+
+import com.luxoft.blockchainlab.corda.hyperledger.indy.flow.AssignPermissionsFlow
+import com.luxoft.blockchainlab.corda.hyperledger.indy.flow.CreatePairwiseFlow
+import com.luxoft.blockchainlab.corda.hyperledger.indy.flow.IssueClaimFlow
+import com.luxoft.blockchainlab.corda.hyperledger.indy.flow.VerifyClaimFlow
+import com.luxoft.blockchainlab.corda.hyperledger.indy.service.IndyService
+import com.luxoft.blockchainlab.hyperledger.indy.utils.PoolManager
+import com.natpryce.konfig.Configuration
+import com.natpryce.konfig.ConfigurationMap
+import com.natpryce.konfig.TestConfigurationsProvider
+import net.corda.core.concurrent.CordaFuture
+import net.corda.core.flows.FlowLogic
+import net.corda.core.identity.CordaX500Name
+import net.corda.core.internal.FlowStateMachine
+import net.corda.core.utilities.getOrThrow
+import net.corda.node.internal.StartedNode
+import net.corda.node.services.api.StartedNodeServices
+import net.corda.testing.common.internal.testNetworkParameters
+import net.corda.testing.node.internal.InternalMockNetwork
+import net.corda.testing.node.internal.InternalMockNetwork.MockNode
+import net.corda.testing.node.internal.newContext
+import org.junit.After
+import org.junit.Before
+import java.util.*
+import kotlin.math.absoluteValue
+
+
+/**
+ * [CordaTestBase] is the base class for any test that uses mocked Corda network.
+ *
+ * Note: [projectServises] and [projectReciverFlows] must be kept updated!
+ * */
+open class CordaTestBase {
+
+    /**
+     * List of all Corda services in the CordApp
+     * */
+    val projectServises = listOf(IndyService::class)
+
+    /**
+     * List of all flows that may be initiated by a message
+     * */
+    val projectReciverFlows = listOf(
+            AssignPermissionsFlow.Authority::class,
+            CreatePairwiseFlow.Issuer::class,
+            IssueClaimFlow.Prover::class,
+            VerifyClaimFlow.Prover::class,
+            VerifyClaimFlow.Prover::class)
+
+    /**
+     * The mocked Corda network
+     * */
+    protected lateinit var net: InternalMockNetwork
+        private set
+
+    private val parties: MutableList<StartedNode<MockNode>> = mutableListOf()
+
+    protected val random = Random()
+
+    /**
+     * Recreate nodes before each test
+     *
+     * Usage:
+     *
+     *     lateinit var issuer: StartedNode<MockNode>
+     *
+     *     @Before
+     *     createNodes() {
+     *         issuer = createPartyNode(CordaX500Name("Issuer", "London", "GB"))
+     *     }
+     * */
+    protected fun createPartyNode(legalName: CordaX500Name? = null): StartedNode<MockNode> {
+        val party = net.createPartyNode(legalName)
```

Comment:
Why legalName is optional?

Reply:
It is optional in Corda, so who am I to change that?

Reply:
We don't need it as optional for our test scenarios.

Reply:
Ok. will change